gtars/src/uniwig/reading.rs

```diff
@@ -0,0 +1,295 @@
+use crate::uniwig::Chromosome;
+use clap::builder::OsStr;
+use flate2::read::GzDecoder;
+use noodles::bam;
+use std::error::Error;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::ops::Deref;
+use std::path::Path;
+
+/// Reads combined bed file from a given path.
+/// Returns Vec of Chromosome struct
+pub fn read_bed_vec(combinedbedpath: &str) -> Vec<Chromosome> {
+    let default_score = 1; // this will later be used for the count, which, by default, was originally = 1
+    let path = Path::new(combinedbedpath);
+
+    let file = File::open(path).unwrap();
+
+    let is_gzipped = path.extension().unwrap_or(&OsStr::from("bed")) == "gz";
+
+    // We must encapsulate in a box and use a dynamic Read trait so that either case could continue.
+    let reader: Box<dyn Read> = match is_gzipped {
+        true => Box::new(GzDecoder::new(file)),
+        false => Box::new(file),
+    };
+
+    let reader = BufReader::new(reader);
+
+    let mut chromosome = Chromosome {
+        chrom: "".to_string(),
+        starts: vec![],
+        ends: vec![],
+    };
+
+    let mut chromosome_vec: Vec<Chromosome> = Vec::new();
+
+    let mut chrom = String::new();
+
+    for line in reader.lines() {
+        //println!("Here is line{:?}", line);
+
+        // Must use a 2nd let statement to appease the borrow-checker
+        let line_string = line.unwrap();
+        let s = line_string.as_str();
+
+        let (parsed_chr, parsed_start, parsed_end) = parse_bed_file(s).unwrap();
+
+        if chrom.is_empty() {
+            // Initial chromosome
+            chromosome.chrom = String::from(parsed_chr.trim());
+            chrom = String::from(parsed_chr.trim());
+            chromosome.starts.push((parsed_start, default_score));
+            chromosome.ends.push((parsed_end, default_score));
+            continue;
+        }
+
+        if String::from(parsed_chr.trim()) != chrom {
```

Comment:
This is creating an owned instance just for comparison. You can use dereferencing:
```rust
if *parsed_chr.trim() != chrom {
```